flybywiresim/aircraft
# feat(a380x): Add "AUTO BRK OFF" ECAM memo, fuel feed tank+ZFW not set faults

## Summary of Changes


- Adds "AUTO BRK OFF" ECAM memo, with preparation for audible alert
- Adds "A/THR OFF" ECAM memo + fault
- Adds "AP OFF" ECAM memo + fault, separate cavalry charge from triple click
- Order MEMOs on EWD and PFD according to importance
- Adds "FUEL FEED TK 1/2/3/4/1+2/3+4 LEVEL LO" ECAM faults
- Adds "FUEL NO ZFW OR ZFWCG DATA" ECAM fault
- Fixes context menu on FMS/FPLN page when no route entered (can open menu now for origin and dest)
- Fixes "NEW DEST" dialog
- ECL: Fix laggy input

## Screenshots (if necessary)



## References

https://www.youtube.com/watch?v=qcXQKQukxkI
amongst others


Discord username (if different from GitHub): floridude

## Testing instructions
Make sure hardware keys for AP and A/THR disconnect have been assigned properly. Avoid something like "AP toggle", that won't work. Choose something like disengage or disconnect. 

1. Disable A/THR using the instinctive A/THR disconnect push button, cancel warning early or only after a few seconds. Memo should be visible on PFD and EWD, along with a single chime
3. Disengage AP using the instinctive AP disconnect push button, cancel warning early or only after a few seconds. Memo should be visible on PFD and EWD, along with cavalry charge and triple click. Master warning p/b should also cancel that. 
4. After landing with basic autobrake mode (LO, 2, 3, HI), deactivate autobrake before reaching 33kts. AUTO BRK OFF memo should be visible on EWD and PFD
5. Empty fuel tanks with EFB below 1500kg per feed tank -> FUEL FEED TK xx LEVEL LO fault should appear on EWD
6. Try taking off without ZFW or ZFWCG filled out in the FMS/FUEL&LOAD page. Maybe you have to clear these fields, they are autofilled if not in cold&dark
7. When starting a flight plan, immediately after entering the city pair manually (no SimBrief), try clicking on origin and destination on the FMS/ACTIVE/F-PLN page and see whether the context menu shows up
8. During a flight, try setting a new destination using the "NEW DEST" option in the F-PLN page context menu
9. During approach with an auto brake mode armed, disconnect A/THR by using the instinctive disconnect push buttons on the thrust levers. Auto brake mode shouldn't be disarmed.


## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page
